### PR TITLE
Feature/ge search bug fix

### LINKED ISSUE #28: Search "Error" Handling
If a user tries to run search without any checked options, there should be a message / it should not redirect

### LINKED ISSUE #40: Fix buggy Search / SearchResults functionality
- [x] Only fetch Applicants upon the click of the Search button
- [x] When you click any button directing away from search-results, it closes the search menu
- [x] When you go "Home" it resets all options to be unselected

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### Type of Change Made 
- [x] bugfix 
- [ ] new feature 
- [ ] breaking change 
- [ ] testing 
- [ ] formatting 
- [x] styling
#### What's this PR do?
* Clears search choices on page-change
* Allows search button to be used multiple times
* Messages user when queries aren't selected
#### Where should the reviewer start?
`Search.tsx` and `Header.tsx`
#### How should this be manually tested?
Click around!
#### Tests are passing
- [ ] 
#### Any background context you want to provide?
* There is potentially a bug if a user goes to another page and reopens the search box too fast, so I made the unmount happen in 500ms. Make me think that offering a "clear" button in the search component could curb that sort of user behavior. 
* Search results still needs to be told not to fetch if it receives an empty query
#### What are the relevant tickets?
close #40, close #28

### CODE REVIEW

File: src/components/Home/Home.tsx [hunk 1]
```diff
@@ -13,11 +13,20 @@ const Home: React.FC = () => {
...
 			<nav className="cta-buttons">
 				<OpenMenuContext.Consumer>
-					{({ toggleMenu }) => (
-						<button className="cta-button" onClick={toggleMenu}>Find Applicants</button>
+					{({isOpen, toggleMenu }) => (
+						<>
+							<button className="cta-button" onClick={toggleMenu}>Find Applicants</button>
+							<Link 
+								to="/" 
+								onClick={() => {
```
Comment (ErinUntermeyer) [inline src/components/Home/Home.tsx:28]: I think that this onClick will need to be moved to the Find Applicants button as that button is no longer opening the menu.

**Review by amykarnaze — APPROVED** [review]
Nice message for letting a user know they need to choose values.

File: src/components/Home/Home.tsx (hunk 1 above)
Comment (amykarnaze) [reply src/components/Home/Home.tsx:28]: Good catch!

Comment (GreysonElkins) [reply src/components/Home/Home.tsx:28]: Nice! Got it in before the merge, that shouldn't be a problem anymore.

**Comment by ErinUntermeyer** [comment]
Awesome work. I'll make a card for implementing the real fetch call when searching for users and note that we need to make sure the fetch does not happen if there isn't at least one option selected.

**Review by ErinUntermeyer — APPROVED** [review]

**Pull request merged**